Code/Core/Math/Random.cpp

```diff
@@ -24,4 +26,15 @@ uint32_t Random::GetRand()
     return( (uint32_t)( m_Seed / ( ( CORE_RAND_MAX + 1 ) * 2 ) ) % ( CORE_RAND_MAX + 1 ) );
 }
 
+// GetRandString
+//------------------------------------------------------------------------------
+uint32_t Random::GetRandString( AString& str )
+{
+    uint32_t rand_num = GetRand();
+    AStackString<128> buf;
+    snprintf( buf.Get(), buf.GetLength(), "%x", rand_num );
```

Comment:
This is a buffer overflow, writing to the "no mans land" of the AStackString.

AString has string formatting functionality built in, so you can replace the 2 lines above and the one below with:
```
str.Format( "%x", randNum );
```

Reply:
I deleted the entire function (as suggested in your next comment).